tools/create-spec

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright © 2022 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import contextlib
+import fileinput
+import json
+import re
+import sys
+
+
+DOWNLOAD_URL = 'https://github.com/cockpit-project/cockpit/releases/download/'
+
+
+def wip_version(args):
+    return 'g' in args.version
+
+
+def substitute(match, args, package):
+    variable = match.group(1)
+
+    if variable == 'VERSION':
+        return args.version
+
+    elif variable == 'RELEASE':
+        return '1.wip' if wip_version(args) else '1'
+
+    elif variable == 'PATH':
+        return '' if wip_version(args) else f'{DOWNLOAD_URL}{args.version}/'
+
+    elif variable == 'BUILD_ALL':
+        return '1' if args.build_all else '0'
+
+    elif variable == 'PATTERNFLY_VERSION':
+        with open('package.json') as fp:
+            return json.load(fp)['dependencies']['patternfly']
+
+    elif variable == 'REQUIRED_BASE':
+        with open(f'pkg/{package}/manifest.json') as fp:
+            return json.load(fp)['requires']['cockpit']
+
+    else:
+        raise ValueError(f'Unsupported substitution @{variable}@')
+
+
+def replace_fields(output, lines, args):
+    variable_re = re.compile(r'@([A-Z_]+)@')
+    package_re = re.compile(r'%package (-n cockpit-)?([a-z]*)')
+    package = None
+
+    for line in lines:
+        if match := package_re.fullmatch(line.strip()):
+            package = match.group(2)
+        line = variable_re.sub(lambda m: substitute(m, args, package), line)
+        output.write(line)
+
+
+@contextlib.contextmanager
+def open_output(filename):
+    if filename:
+        fp = open(filename, 'w+')
+        yield fp
+        fp.close()
+    else:
+        yield sys.stdout
```

Comment:
FWIW, I find this too complicated. I'd just drop this function completely, and do
```py
with (open(args.output, "w+") if args.output else sys.stdout) as output:
```
but *shrug*, not a biggie.

Reply:
I agree that this is ridiculous.  I think i was trying to invent the analogue of `fileinput` for output.

Your solution has the side-effect of closing stdout, which seems like something we shouldn't do.

I don't have any better ideas, so I'll just commit this one :)